docs/products/kafka/howto/connect-with-java.rst

```diff
@@ -61,64 +61,115 @@ Variable                     Description
-    properties.put("ssl.keystore.type", "PKCS12");
-    properties.put("ssl.keystore.location", "{KEYSTORE_LOCATION}");
-    properties.put("ssl.keystore.password", "{KEYSTORE_PASSWORD}");
-    properties.put("ssl.key.password", "{KEY_PASSWORD}");
-    properties.put("key.serializer", "{SERIALIZER}");
-    properties.put("value.serializer", {SERIALIZER}");
-    properties.put("key.deserializer", "{DESERIALIZER}");
-    properties.put("value.deserializer", "{DESERIALIZER}");
-
-    // create a producer
-    KafkaProducer<String, String> producer = new KafkaProducer<>(properties);
-    // create a consumer
-    KafkaConsumer<String, String> consumer = new KafkaConsumer<>(properties);
-
-With SASL-SSL authentication
------------------------------
-
-Set up properties to connect to the cluster and create a producer and a consumer:
+With SSL authentication
+~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. code::
 
-    Properties properties = new Properties();
+        Properties properties = new Properties();
+        properties.put("bootstrap.servers", "{HOST}:{SSL_PORT}");
+        properties.put("security.protocol", "SSL");
+        properties.put("ssl.truststore.location", "{TRUSTSTORE_LOCATION}");
+        properties.put("ssl.truststore.password", "{TRUSTSTORE_PASSWORD}");
+        properties.put("ssl.keystore.type", "PKCS12");
+        properties.put("ssl.keystore.location", "{KEYSTORE_LOCATION}");
+        properties.put("ssl.keystore.password", "{KEYSTORE_PASSWORD}");
+        properties.put("ssl.key.password", "{KEY_PASSWORD}");
+        properties.put("key.serializer", "{SERIALIZER}");
+        properties.put("value.serializer", "{SERIALIZER}");
+        properties.put("key.deserializer", "{DESERIALIZER}");
+        properties.put("value.deserializer", "{DESERIALIZER}");
+
+        // create a producer
+        KafkaProducer<String, String> producer = new KafkaProducer<>(properties);
+
+With SASL authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::    
+      
+        String sasl_username = "{USER_NAME}";
+        String sasl_password = "{SASL_PASSWORD}";
+        String jaasTemplate = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"%s\" password=\"%s\";";
+        String jaasConfig = String.format(jaasTemplate, sasl_username, sasl_password);
+          
+        Properties properties = new Properties();
+        properties.put("bootstrap.servers", "{HOST}:{SASL_PORT}");
+        properties.put("security.protocol", "SASL_SSL");
+        properties.put("sasl.mechanism", "SCRAM-SHA-256");
+        properties.put("sasl.jaas.config", jaasConfig);
+        properties.put("ssl.endpoint.identification.algorithm", ""); 
+        properties.put("ssl.truststore.type", "jks");
+        properties.put("ssl.truststore.location", "{TRUSTSTORE_LOCATION}");
+        properties.put("ssl.truststore.password", "{TRUSTSTORE_PASSWORD}");
+        properties.put("key.serializer", "{SERIALIZER}");
```

Comment:
Since we divided producers and consumers, I believe we can also keep only serialised for producers and deserialiser for consumers

Reply:
Good point. Fixed.